### PR TITLE
Partial move to gradle kotlin

### PR DESCRIPTION
**Pull request by yschimke**
Move the base functionality without tackling the most difficult parts such that an incremental migration is possible.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,67 +1,9 @@
...
-      'equinox': "org.eclipse.platform:org.eclipse.osgi:${versions.equinox}",
-      'kotlinStdlibOsgi': "org.jetbrains.kotlin:kotlin-osgi-bundle:${versions.kotlin}"
-  ]
-
   dependencies {
     classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.21"
     classpath "org.jetbrains.dokka:dokka-gradle-plugin:0.10.1"
     classpath "com.android.tools.build:gradle:4.0.2"
-    classpath deps.bnd
+    classpath Dependencies.bnd
```
Comment (swankjesse) [inline build.gradle:6]: Oooh NICE

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
```
Comment (swankjesse) [inline buildSrc/build.gradle.kts:2]: 2021 !

File: buildSrc/src/main/kotlin/artifacts.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  /** Returns the artifact ID for the project, or null if it is not published. */
+  @JvmStatic
+  fun publishedArtifactId(projectName: String): String? {
+    return when (projectName) {
+      "okhttp-logging-interceptor" -> "logging-interceptor"
+      "mockwebserver" -> "mockwebserver3"
+      "mockwebserver-junit4" -> "mockwebserver3-junit4"
+      "mockwebserver-junit5" -> "mockwebserver3-junit5"
+      "mockwebserver-deprecated" -> "mockwebserver"
+      in listOf(
```
Comment (swankjesse) [inline buildSrc/src/main/kotlin/artifacts.kt:11]: omit this line, and I think you get the same behavior?

**Review by swankjesse — APPROVED** [review]
I love it.

File: buildSrc/src/main/kotlin/alpn.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
+object Alpn {
```
Comment (yschimke) [inline buildSrc/src/main/kotlin/alpn.kt:1]: copyright to come.

**Pull request merged**